File: 1502-construct-k-palindrome-strings/1502-construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        if(s.length()<k){
            return false;
        }
        Map<Character,Integer> hash=new HashMap<>();
        for(char c:s.toCharArray()){
            hash.put(c,hash.getOrDefault(c,0)+1);
        }
        ArrayList<Integer> val=new ArrayList<>(hash.values());
        int count=0;
        for(int i:val){
            if(i%2==1){
                count++;
            }
        }
        if(count<=k){
            return true;
        }
        else{
            return false;
        }
    }
}

File: 1502-construct-k-palindrome-strings/1502-construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if len(s)<k:
            return False
        hash=Counter(s)
        print(hash)
        count=0
        for v in hash.values():
            if v%2==1:
                count+=1
        return True if count<=k else False
